azure.go

```diff
@@ -136,6 +142,23 @@ func (p *azureProvider) VMSSClient(subscriptionID string) (vmssClient, error) {
 }
 
 func (p *azureProvider) getAuthorizer() (autorest.Authorizer, error) {
+	p.lock.RLock()
+	unlockFunc := p.lock.RUnlock
+	defer func() { unlockFunc() }()
```

Comment:
I don't think this is worth doing, I would just release the read lock inside the if block before returning.  It's the very next line after all.

Reply:
Or rather, RLock, fetch p.authorizorExpiration, RUnlock, then if/return/etc.

Reply:
This is a pattern we've used a few places where an upgrade is necessary. I've carried it forward mainly b/c of precedent, but I do like getting the deferred unlock out of the way early so there is no question later (more an issue for a longer function that might pick up another return).